lib/shared/text-table.js

```diff
@@ -0,0 +1,110 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE
+ */
+
+"use strict";
+
+/**
+ *
+ * @param c
+ */
+function dotindex(c) {
+    const m = /\.[^.]*$/u.exec(c);
+
+    return m ? m.index + 1 : c.length;
+}
+
+module.exports = function(rows_, opts) {
+    const hsep = "  ";
+    const align = opts.align;
+    const stringLength = opts.stringLength;
+    const dotsizes = rows_.reduce((acc, row) => {
+        row.forEach((c, ix) => {
+            const n = dotindex(c);
+
+            if (!acc[ix] || n > acc[ix]) {
+                acc[ix] = n;
+            }
+        });
+        return acc;
+    }, []);
+
+    const rows = rows_.map(row =>
+        row.map((c_, ix) => {
+            const c = String(c_);
+
+            if (align[ix] === ".") {
+                const index = dotindex(c);
+                const size =
+                    dotsizes[ix] +
+                    (/\./u.test(c) ? 1 : 2) -
+                    (stringLength(c) - index);
+
+                return c + Array(size).join(" ");
+            }
+            return c;
+        }));
+
+    const sizes = rows.reduce((acc, row) => {
+        row.forEach((c, ix) => {
+            const n = stringLength(c);
+
+            if (!acc[ix] || n > acc[ix]) {
+                acc[ix] = n;
+            }
+        });
+        return acc;
+    }, []);
+
+    return rows
+        .map(row =>
+            row
+                .map((c, ix) => {
+                    const n = sizes[ix] - stringLength(c) || 0;
+                    const s = Array(Math.max(n + 1, 1)).join(" ");
+
+                    if (align[ix] === "r" || align[ix] === ".") {
+                        return s + c;
+                    }
+                    if (align[ix] === "c") {
+                        return (
+                            Array(Math.ceil(n / 2 + 1)).join(" ") +
+                            c +
+                            Array(Math.floor(n / 2 + 1)).join(" ")
+                        );
+                    }
```

Comment:
Since `align` is never `'c'`, and `s + c` is the same as `c + s`:

```suggestion
```

Reply:
Removed, and we no longer need to pass `align` as an option too.

Reply:
It seems we need the `s+c` check, otherwise regression as reported in https://github.com/eslint/eslint/pull/19061#pullrequestreview-2411621760